### PR TITLE
Don't generate `SerialNo` attribute for anything except the MainProduct during Manufacturing

### PR DESCRIPTION
**Pull request by TheBestPessimist**
https://github.com/metasfresh/metasfresh/issues/10426

### CODE REVIEW

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/pporder/api/impl/AbstractPPOrderReceiptHUProducer.java [hunk 1]
```diff
@@ -310,7 +310,7 @@ public final void createReceiptCandidatesFromPlanningHU(@NonNull final I_M_HU pl
 			}
 
 			//
-			updateReceivedHUs(huContext, ImmutableSet.of(planningHU));
+			updateReceivedHUs(huContext, ImmutableSet.of(planningHU), null);  // TODO tbp: not sure if this should be null
```
Comment (teosarca) [inline backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/pporder/api/impl/AbstractPPOrderReceiptHUProducer.java:313]: consider removing createReceiptCandidatesFromPlanningHU() at all.
in some branches, I've already removed.
so u don't have to bother fixing this TODO

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/pporder/api/impl/AbstractPPOrderReceiptHUProducer.java (hunk 1 above)
Comment (TheBestPessimist) [reply backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/pporder/api/impl/AbstractPPOrderReceiptHUProducer.java]: removed the method from interface and implementation.

Comment (TheBestPessimist) [reply backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/pporder/api/impl/AbstractPPOrderReceiptHUProducer.java]: done

**Review by teosarca — APPROVED** [review]

**Pull request merged**